core/src/main/java/com/linecorp/armeria/common/multipart/MimeParser.java

```diff
@@ -415,38 +415,78 @@ private static ByteBuf safeReadBytes(StreamDecoderInput in, int length) {
      * Skips the preamble.
      */
     private void skipPreamble() {
-        // matches boundary
-        boundaryStart = match();
-        if (boundaryStart == -1) {
+        boundaryStart = -1;
+        final int boundaryStartOffset = match();
+        if (boundaryStartOffset == -1) {
             // No boundary is found
             return;
         }
 
         final int length = in.readableBytes();
 
-        // Consider all the whitespace boundary+whitespace+"\r\n"
+        // Consider all the whitespace. e.g. boundary+whitespace+"\r\n"
         int linearWhiteSpace = 0;
-        for (int i = boundaryStart + boundaryLength;
+        for (int i = boundaryStartOffset + boundaryLength;
              i < length && (in.getByte(i) == ' ' || in.getByte(i) == '\t'); i++) {
             ++linearWhiteSpace;
         }
 
-        // Check for \n or \r\n
-        if (boundaryStart + boundaryLength + linearWhiteSpace < length &&
-            (in.getByte(boundaryStart + boundaryLength + linearWhiteSpace) == '\n' ||
-             in.getByte(boundaryStart + boundaryLength + linearWhiteSpace) == '\r')) {
+        // Three valid cases:
+        // boundary+whitespace+"\n"
+        // boundary+whitespace+"\r\n"
+        // boundary+whitespace+"--" + "whatever after --"
```

Comment:
Question: If the third case means a closing boundary, shouldn't `whitespace` be considered?
For example:
```suggestion
        // boundary+"--" + "whatever after --"
```

Reply:
That's a good point. 👍